src/DynamoCore/Core/NodeModelAssemblyLoader.cs

```diff
@@ -199,6 +199,44 @@ internal static bool ContainsNodeModelSubType(Assembly assem)
             return assem.GetTypes().Any(IsNodeSubType);
         }
 
+        internal static bool ContainsNodeModelSubTypeReflectionLoaded(Assembly assem, Type nodeModelType = null)
+        {
+            if(!(nodeModelType is null))
+                return assem.GetTypes().Any(x => IsNodeSubTypeReflectionLoaded(x, nodeModelType));
+            return assem.GetTypes().Any(x => IsNodeSubType(x));
+        }
+
+        internal static bool IsNodeSubTypeReflectionLoaded(Type t, Type nodeModelType)
+        {
+            bool isNodeSubType = false;
+
+            try
+            {
+                isNodeSubType = !t.IsAbstract &&
+                    t.IsSubclassOf(nodeModelType) &&
+                    t.GetConstructor(Type.EmptyTypes) != null;
```

Comment:
From this I read that when trying to get a parameterless ctor when there's none it will throw an exception?
Might be better to use a linq query like `t.GetConstructors().FirstOrDefault(ctor => ctor.GetParameters().Lengh == 0) != null`, as it will reduce memory allocations when throwing an exception

Reply:
From my understanding, is better to avoid raising an exception if possible as when doing so the exception collects a snapshot of the state of the application (stacktrace and all that stuff).

I don't know how it compares in performance between reflection vs throwing exceptions, but using reflection it explicitly tells what the intent.